### PR TITLE
[FEATURE] DataAssistants Example Notebook - Pandas

### PR DESCRIPTION
**Pull request by Shinnnyshinshin**
Changes proposed in this pull request:
- Update `tests/test_fixtures/rule_based_profiler/example_notebooks/DataAssistants_Instantiation_And_Running-OnboardingAssistant-Pandas.ipynb` to include the most recent workflow (which was first shown in the sql `DataAssistants` example notebook)
### The Workflow looks like this 
- Running DataAssistant on 2019 Taxi Data (our "training set")
- Running Checkpoint on 2020 January Taxi Data (our "test set")
- Update Appendix to include parameter descriptions for `DataAssistant`, and how they can be adjusted / modified
- **Note**: Since `Pandas` does not currently support being able to define a schema, there is a note that describes this limitation. Schemas are supported by the sql example (in the database), and `Spark` (soon) 

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

### CODE REVIEW

File: tests/test_fixtures/rule_based_profiler/example_notebooks/DataAssistants_Instantiation_And_Running-OnboardingAssistant-Pandas.ipynb [hunk 1]
```diff
@@ -0,0 +1,911 @@
...
+    "from great_expectations.validator.validator import Validator\n",
+    "from great_expectations.rule_based_profiler.data_assistant import (\n",
+    "    DataAssistant,\n",
+    "    VolumeDataAssistant,\n",
+    ")\n",
+    "from great_expectations.rule_based_profiler.data_assistant_result import (\n",
+    "    VolumeDataAssistantResult,\n",
+    ")\n",
+    "from typing import List\n",
+    "from ruamel import yaml\n"
```
Comment (anthonyburdi) [inline tests/test_fixtures/rule_based_profiler/example_notebooks/DataAssistants_Instantiation_And_Running-OnboardingAssistant-Pandas.ipynb]: Maybe we can use YAMLHandler here?
```suggestion
    "from great_expectations.core.yaml_handler import YAMLHandler\n",
    "yaml = YAMLHandler()\n"
```

File: tests/test_fixtures/rule_based_profiler/example_notebooks/DataAssistants_Instantiation_And_Running-OnboardingAssistant-Pandas.ipynb [hunk 2]
```diff
@@ -0,0 +1,911 @@
...
+   "execution_count": null,
+   "id": "4857e285",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# import shutil, os\n",
+    "# shutil.rmtree(f\"great_expectations/expectations/\")\n",
+    "# shutil.rmtree(f\"great_expectations/checkpoints/\")\n",
+    "# shutil.rmtree(f\"great_expectations/uncommitted/\")\n",
+    "# os.remove(\"great_expectations/expectations/.ge_store_backend_id\")\n"
```
Comment (anthonyburdi) [inline tests/test_fixtures/rule_based_profiler/example_notebooks/DataAssistants_Instantiation_And_Running-OnboardingAssistant-Pandas.ipynb]: We should leave the /expectations/ folder and the .ge_store_backend_id to make sure that we don't inadvertently create a new data_context_id when this user continues working with the library. Also why are we deleting checkpoints and uncommitted?

**Review by anthonyburdi — CHANGES_REQUESTED** [review]
Just a few requested changes, but otherwise looks great! Thank you for creating this :)

File: tests/test_fixtures/rule_based_profiler/example_notebooks/DataAssistants_Instantiation_And_Running-OnboardingAssistant-Pandas.ipynb (hunk 1 above)
Comment (Shinnnyshinshin) [reply tests/test_fixtures/rule_based_profiler/example_notebooks/DataAssistants_Instantiation_And_Running-OnboardingAssistant-Pandas.ipynb]: done!

File: tests/test_fixtures/rule_based_profiler/example_notebooks/DataAssistants_Instantiation_And_Running-OnboardingAssistant-Pandas.ipynb (hunk 2 above)
Comment (Shinnnyshinshin) [reply tests/test_fixtures/rule_based_profiler/example_notebooks/DataAssistants_Instantiation_And_Running-OnboardingAssistant-Pandas.ipynb]: Great point. I think the original intent was to "clean up" the files that were generated when the notebook is run, but this was a bit like taking a sledge hammer to the `great_expectations/` folder.  I'm going to see if I can make this more precise :)

Comment (Shinnnyshinshin) [reply tests/test_fixtures/rule_based_profiler/example_notebooks/DataAssistants_Instantiation_And_Running-OnboardingAssistant-Pandas.ipynb]: The section now only removes the Suite and Checkpoint that was created
``` 
import os
os.remove('great_expectations/checkpoints/my_checkpoint.yml')
os.remove('great_expectations/expectations/taxi_data_2019_suite.json')
```

**Review by anthonyburdi — APPROVED** [review]
LGTM!
